raphtory/src/vectors/vectorised_graph.rs

```diff
@@ -85,452 +70,169 @@ impl<G: StaticGraphViewOps, T: DocumentTemplate<G>> VectorisedGraph<G, T> {
-    /// hops away of  each other if there is a document B such that A is n - 1 hops away of B and B
-    /// is 1 hop away of C.
-    ///
-    /// # Arguments
-    ///   * hops - the number of hops to carry out the expansion
-    ///   * window - the window where documents need to belong to in order to be considered
-    ///
-    /// # Returns
-    ///   A new vectorised graph containing the updated selection
-    pub fn expand(&self, hops: usize, window: Option<(i64, i64)>) -> Self {
-        match window {
-            None => self.expand_with_window(hops, window, &self.source_graph),
-            Some((start, end)) => {
-                let windowed_graph = self.source_graph.window(start, end);
-                self.expand_with_window(hops, window, &windowed_graph)
-            }
-        }
-    }
-
-    fn expand_with_window<W: StaticGraphViewOps>(
-        &self,
-        hops: usize,
-        window: Option<(i64, i64)>,
-        windowed_graph: &W,
-    ) -> Self {
-        let mut selected_docs = self.selected_docs.clone();
-        for _ in 0..hops {
-            let context = selected_docs
-                .iter()
-                .flat_map(|(doc, _)| self.get_context(doc, windowed_graph, window))
-                .map(|doc| (doc.clone(), 0.0));
-            selected_docs = extend_selection(selected_docs.clone(), context, usize::MAX);
-        }
-
-        Self {
-            selected_docs,
-            ..self.clone()
-        }
+    ) -> VectorSelection<G> {
+        let joined = chain!(self.node_documents.iter(), self.edge_documents.iter());
+        let docs = self.search_top_document_groups(joined, query, limit, window);
+        VectorSelection::new_with_preselection(self.clone(), docs)
     }
 
-    /// Add the top `limit` adjacent documents with higher score for `query` to the selection
-    ///
-    /// The expansion algorithm is a loop with two steps on each iteration:
-    ///   1. All the documents 1 hop away of some of the documents included on the selection (and
-    /// not already selected) are marked as candidates.
-    ///  2. Those candidates are added to the selection in descending order according to the
-    /// similarity score obtained against the `query`.
-    ///
-    /// This loops goes on until the current selection reaches a total of `limit`  documents or
-    /// until no more documents are available
+    /// Search the top scoring nodes according to `query` with no more than `limit` nodes
     ///
     /// # Arguments
-    ///   * query - the text or the embedding to score against
+    ///   * query - the embedding to score against
+    ///   * limit - the maximum number of nodes to search
     ///   * window - the window where documents need to belong to in order to be considered
     ///
     /// # Returns
-    ///   A new vectorised graph containing the updated selection
-    pub fn expand_by_similarity(
+    ///   The vector selection resulting from the search
+    pub fn search_nodes(
```

Comment:
rename to nodes_by_similarity, also in python, the rest of the functions as well